### PR TITLE
[5.2] Adds operators to where

### PR DESCRIPTION
**Pull request by mlantz**
Adds operators to where in collections, makes finding stuff much easier.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -206,13 +206,30 @@ public function filter(callable $callback = null)
...
+        return $this->filter(function ($item) use ($key, $value, $strict, $operator) {
+            $itemKey = data_get($item, $key);
+            if ($strict && ($operator == '!=' || $operator == '!==')) {
+                $operator = '!==';
+            }
+            if ($strict && ($operator == '==' || $operator == '===')) {
+                $operator = '===';
+            }
+            switch ($operator) {
+                case '==':  return $itemKey ==  $value;
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: Please format these according to psr-2.

**Comment by JosephSilber** [comment]
Oh please! This is just silly :expressionless: 

If we are to do this, we should do it [the way I originally did it](https://github.com/laravel/framework/pull/6502).

It would be breaking to introduce that now, so we should do it in 5.3

**Comment by JosephSilber** [comment]
I actually went ahead and submitted a new PR: https://github.com/laravel/framework/pull/12429

**Comment by chrisblackwell** [comment]
@JosephSilber Why would you say this is just silly, and then put in a new pull request, doing the same thing? Why not provide some feedback to @mlantz on what could change? This seems very petty.

**Comment by barryvdh** [comment]
This PR: `->where('stars', 5, true, '>');`. The PR by @JosephSilber uses: `->where('stars', '>', 5);`. Makes more sense to me :)
(Also, he suggested this in Nov 2014 for L5.0, so it's not like he's stealing his idea..)

**Comment by chrisblackwell** [comment]
@barryvdh It's not about stealing his idea. It's about the 

> Oh please! This is just silly :expressionless:

Why make a person feel bad about something they've tried to share, just because you did it before?

**Comment by JosephSilber** [comment]
Maybe that was not so nice. I deleted that part from my comment. Sorry @mlantz

**Comment by mlantz** [comment]
I appreciate it, thanks @JosephSilber

**Pull request closed without merging**